docs/basics/networkbehaviour.md

```diff
@@ -108,3 +108,130 @@ If you override the virtual 'OnDestroy' method it is important to alway invoke t
+    /// server-side m_ToggleState
+    /// </summary>
+    /// <param name="stateIsSet"></param>
+    [ClientRpc]
+    private void ToggleStateClientRpc(bool stateIsSet)
+    {
+        m_ToggleState = stateIsSet;
+    }
+}
+```
+:::caution
+Since `NetworkBehaviour.OnSynchronize` is primarily used for server to client synchronization, RPC state synchronization only works when using ClientRpcs since `NetworkBehaviour.OnSynchronize` is only invoked on the server side during the write portion of serialization and only invoked on the client side during the read portion of serialization. When running a host `NetworkBehaviour.OnSynchronize` is still only invoked once (server-side) during the write portion of serialization.
+:::
+
+### Debugging OnSynchronize Serialization
+If your serialization code has a bug and throws an exception, then `NetworkBehaviour.OnSynchronize` has additional safety checking to handle a graceful recovery without completely breaking the rest of the synchronization serialization pipeline.  
+
+#### When Writing:
+If user-code throws an exception during `NetworkBehaviour.OnSynchronize`, it catches the exception and if:
+- **LogLevel = Normal**: A warning message that includes the name of the `NetworkBehaviour` that threw an exception while writing will be logged and that portion of the serialization for the given `NetworkBehaviour` is skipped.
+- **LogLevel = Developer**: It provides the same warning message as well as it logs an error with the exception message and stack trace.
+
+After generating the log message(s), it rewinds the serialization stream to the point just before it invoked `NetworkBehaviour.OnSynchronize` and will continue serializing. Any data written before the exception occurred will be overwritten or dropped depending upon whether there are more `NetworkBehaviour` components to be serialized.
+
+### When Reading:
+For exceptions this follows the exact same message logging pattern described above when writing. The distinct difference is that after it logs one or more messages to the console, it skips over only the serialization data written by the server-side when `NetworkBehaviour.OnSynchronize` was invoked and continues the deserialization process for any remaining `NetworkBehaviour` components.
+
+However, there is an additional check to assure that the total expected bytes to read were actually read from the buffer. If the total number of bytes read does not equal the expected number of bytes to be read it will log a warning that includes the name of the NetworkBehaviour in question, the total bytes read, the expected bytes to be read, and lets you know this `NetworkBehaviour` is being skipped.
+
+:::caution
+When using `NetworkBehaviour.OnSynchronize` you should be aware that you are increasing the synchronization payload size per instance.  If you have 30 instances that each write 100 bytes of information you will have increased the total full client synchronization size by 3000 bytes
+:::
```

Comment:
For exceptions, this follows the exact same message-logging pattern described above when writing. The distinct difference is that after it logs one or more messages to the console, it skips over only the serialization data written by the server-side when `NetworkBehaviour.OnSynchronize` was invoked and continues the deserialization process for any remaining `NetworkBehaviour` components.

However, there is an additional check to ensure the total expected bytes to read were actually read from the buffer. If the total number of bytes read doesn't equal the expected number of bytes, it logs a warning that includes the name of the `NetworkBehaviour` in question, the total bytes read, and the expected bytes to read. It also lets you know the `NetworkBehaviour` is being skipped.

:::caution
When using `NetworkBehaviour.OnSynchronize`, you increase the synchronization payload size per instance. If you have 30 instances that each write 100 bytes of information, you increase the total full client synchronization size by 3000 bytes.
:::